WireDomain/Sources/WireDomain/Storage/UserDefaultsWrapper.swift

```diff
@@ -0,0 +1,93 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Combine
+import Foundation
+
+/// Allows to match for optionals with generics that are defined as non-optional.
+protocol AnyOptional {
+    /// Returns `true` if `nil`, otherwise `false`.
+    var isNil: Bool { get }
+}
+
+extension Optional: AnyOptional {
```

Comment:
I have seen this kind of syntactic sugar in the code base and other developers joining fighting against it ending removing it for clarity

Reply:
This is needed by our propertyWrapper to know whether or not we're dealing with an optional so we can call the `container.removeObject(forKey: scopeKey(key))`  method consequently. This is not exposed by WireDomain, I could make the protocol `private` to enforce that.